File: BackEnd/src/domain/dtos/user/change-password-request.dto.ts
/**
 * @file change-password-request.dto.ts
 * @description DTO de cambio de contraseña. Se completará en Fase 2.
 * @module User
 * @layer Domain
 */
import { IsString, MinLength } from 'class-validator';

export class ChangePasswordRequestDto {
  @IsString() currentPassword!: string;
  @IsString() @MinLength(8) newPassword!: string;
}

File: FrontEnd/lib/domain/dtos/user/change_password_request_dto.dart
/// @file change_password_request_dto.dart
/// @description DTO para cambio de contraseña del usuario.
/// @module User
/// @layer Domain
library;

// ═══════════════════════════════════════════════════════════════════════════════
// CHANGE PASSWORD REQUEST DTO
// ═══════════════════════════════════════════════════════════════════════════════

/// DTO de cambio de contraseña.
class ChangePasswordRequestDto {
  const ChangePasswordRequestDto({
    required this.currentPassword,
    required this.newPassword,
  });

  final String currentPassword;
  final String newPassword;

  Map<String, dynamic> toJson() => {
    'currentPassword': currentPassword,
    'newPassword':     newPassword,
  };
}
